news/tasks.py

```diff
@@ -340,28 +357,13 @@ def upsert_user(api_call_type, data):
     @param dict data: POST data from the form submission
     @return:
     """
-    if DO_LOCKING:
-        lock_key = data.get('email') or data.get('token')
-        lock_key = sha256(lock_key).hexdigest()
-        lock = cache.lock(lock_key, timeout=60)
-        if lock.acquire(blocking=False):
-            try:
-                upsert_contact(api_call_type, data,
-                               get_user_data(data.get('token'), data.get('email'),
-                                             extra_fields=['id']))
-            finally:
-                try:
-                    lock.release()
-                except LockError:
-                    pass
-        else:
-            statsd.incr('news.tasks.upsert_user.no_lock_retry')
-            raise RetryTask
+    email = data.get('email')
```

Comment:
In thinking about it I think it's really only needed for email address. The purpose of the locking is to avoid duplicate records, and the only time that should occur is when we're dealing only with an email address and need to look up an account. If we're dealing with a token then they've been added by email at some point previous. It wouldn't hurt to lock by token though...